src/extractor/ConfigurationExtractor.ts

```diff
@@ -59,7 +59,15 @@ export class ConfigurationExtractor {
     const canonicalMatch = canonical.match(/^https?:\/\/([\w.]+)(\/.*)?$/);
     if (canonicalMatch) {
       const hostParts = canonicalMatch[1].split('.').slice(0, -1);
-      const pathParts = canonicalMatch[2]?.split('/') ?? [];
+      const pathParts = canonicalMatch[2]?.split('/').filter(part => part.length > 0) ?? [];
+      // Many IGs are hosted at a URL that starts with http://hl7.org/fhir/
+      // However, they generally do not include "hl7" and "fhir" at the start of their id and name.
```

Comment:
I'm afraid you've taken this a bit too far!  While they don't include "hl7" or "fhir" in their `name`, they _do_ include them in their `id` (and, in fact, I think the IG Publisher requires them in the `id` if they are in the `canonical`).

So this change really should only apply to the `name`.